cmd/kube-proxy/app/server_test.go

```diff
@@ -396,3 +415,36 @@ func TestLoadConfigFailures(t *testing.T) {
 		}
 	}
 }
+
+func TestKubeProxyVersionUpdated(t *testing.T) {
+	node1 := clientv1.Node{ObjectMeta: metav1.ObjectMeta{Name: "node-1"}}
+	fakeClient := fake.NewSimpleClientset(&node1)
+	nodePatched := make(chan struct{})
+	fakeClient.AddReactor("patch", "nodes", func(action clienttesting.Action) (bool, k8sRuntime.Object, error) {
+		if action.GetSubresource() == "status" {
+			patchAction := action.(clienttesting.PatchActionImpl)
+			if patchAction.Name == node1.ObjectMeta.Name {
+				close(nodePatched)
+			}
+			return true, nil, nil
+		}
+		return false, nil, nil
+	})
+	server := &ProxyServer{
+		Client:                fakeclientset.NewSimpleClientset(),
+		CoreClient:            fakeClient.Core(),
+		EndpointsEventHandler: &fakeEndpointsHandler{},
+		NodeRef: &clientv1.ObjectReference{
+			Name: node1.ObjectMeta.Name,
+		},
+		Proxier:             &fakeProxyProvider{},
+		Recorder:            &record.FakeRecorder{},
+		ServiceEventHandler: &fakeServiceHandler{},
+	}
+	server.Run()
+	select {
+	case <-nodePatched:
+	case <-time.After(time.Second):
```

Comment:
Trust me, you'll be happier using a longer timeout to avoid potential flakes due to cpu contention :-)